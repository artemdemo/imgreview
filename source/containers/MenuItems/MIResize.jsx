/* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for,react/no-unused-state */
import React from 'react';
import { connect } from 'react-redux';
import { Form, Field } from 'react-final-form';
import Icon from '../../components/Icon/Icon';
import Popup from '../../components/Popup/Popup';
import PopupButtonsContainer from '../../components/Popup/PopupButtonsContainer';
import Button from '../../components/Button/Button';
import FormGroup from '../../components/FormGroup/FormGroup';
import FormInput from '../../components/FormInput/FormInput';
import FormButtonsRow from '../../components/FormButtonsRow/FormButtonsRow';
import MainMenuItem from '../../components/MainMenu/MainMenuItem';
import { couldBeNumber } from '../../services/number';
import { updateImageSize } from '../../model/canvas/canvasActions';

class MIResize extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            width: '0',
            height: '0',
        };

        this.popupRef = React.createRef();
    }

    onClick = () => {
        this.popupRef.current.show();
    };

    onCancel = () => {
        this.popupRef.current.hide();
    };

    onPopupOpen = () => {
        const { canvas } = this.props;
        const { width, height } = canvas.image.getSize();
        this.setState({
            width,
            height,
        });
    };

    onSubmit = (values) => {
        const { updateImageSize } = this.props;
        const width = Number(values.width);
        const height = Number(values.height);
        if (width > 0 && height > 0) {
            updateImageSize({width, height});
            this.popupRef.current.hide();
        }
    };

    // ToDo: Since I started to use final-form this method is not in use
    //  but I want to find a way to use it.
    // updateSize(sizeKey, e) {
    //     const { value } = e.target;
    //     const secondSizeKey = sizeKey === 'width' ? 'height' : 'width';
    //
    //     const calcSecondSize = () => {
    //         if (value === '') {
    //             return '';
    //         }
    //         const numValue = Number(value);
    //         const ratio = this.state[`${secondSizeKey}Init`] / this.state[`${sizeKey}Init`];
    //         return Math.round(numValue * ratio);
    //     };
    //
    //     if (couldBeNumber(value) || value === '') {
    //         this.setState({
    //             [sizeKey]: value,
    //             [secondSizeKey]: calcSecondSize(),
    //         });
    //     }
    // }

    render() {
        const { canvas } = this.props;
        return (
            <React.Fragment>
                <MainMenuItem
                    onClick={this.onClick}
                    disabled={canvas.image == null}
                >
                    <Icon
                        name='expand'
                        title='Resize'
                    />
                </MainMenuItem>
                <Popup
                    title='Resize image'
                    ref={this.popupRef}
                    onOpen={this.onPopupOpen}
                    showCloseBtn={false}
                >
                    <Form
                        initialValues={this.state}
                        onSubmit={this.onSubmit}
                        validate={(values) => {
                            const errors = {};
                            if (!couldBeNumber(values.width)) {
                                errors.width = 'Must be a number';
                            } else if (Number(values.width) > 5000) {
                                errors.width = 'Value is too big';
                            }
                            if (!couldBeNumber(values.height)) {
                                errors.height = 'Must be a number';
                            } else if (Number(values.height) > 5000) {
                                errors.height = 'Value is too big';
                            }
                            return errors;
                        }}
                        render={({ handleSubmit, invalid, form }) => (
                            <form onSubmit={handleSubmit}>
                                <div className='row'>
                                    <div className='col-sm'>
                                        <Field
                                            name='width'
                                            render={({ input, meta }) => (
                                                <FormGroup
                                                    errorText={meta.error}
                                                >
                                                    <label htmlFor='img-width'>Width (px)</label>
                                                    <FormInput
                                                        placeholder='Enter width'
                                                        id='img-width'
                                                        {...input}
                                                    />
                                                </FormGroup>
                                            )}
                                        />
                                    </div>
                                    <div className='col-sm'>
                                        <Field
                                            name='height'
                                            render={({ input, meta }) => (
                                                <FormGroup
                                                    errorText={meta.error}
                                                >
                                                    <label htmlFor='img-height'>Height (px)</label>
                                                    <FormInput
                                                        placeholder='Enter height'
                                                        id='img-height'
                                                        {...input}
                                                    />
                                                </FormGroup>
                                            )}
                                        />
                                    </div>
                                </div>
                                <PopupButtonsContainer>
                                    <FormButtonsRow>
                                        <Button
                                            onClick={this.onCancel}
                                            secondary
                                        >
                                            Cancel
                                        </Button>
                                        <Button
                                            type='submit'
                                            disabled={invalid}
                                        >
                                            Resize
                                        </Button>
                                    </FormButtonsRow>
                                </PopupButtonsContainer>
                            </form>
                        )}
                    />
                </Popup>
            </React.Fragment>
        );
    }
}

export default connect(
    state => ({
        canvas: state.canvas,
    }), {
        updateImageSize,
    },
)(MIResize);
