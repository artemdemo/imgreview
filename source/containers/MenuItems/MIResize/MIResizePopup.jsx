/* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import { Field, Form, FormSpy } from 'react-final-form';
import FormGroup from '../../../components/FormGroup/FormGroup';
import FormInput from '../../../components/FormInput/FormInput';
import PopupButtonsContainer from '../../../components/Popup/PopupButtonsContainer';
import FormButtonsRow from '../../../components/FormButtonsRow/FormButtonsRow';
import Button from '../../../components/Button/Button';
import Popup from '../../../components/Popup/Popup';
import { couldBeNumber } from '../../../services/number';

class MIResizePopup extends React.PureComponent {
    static validate(values) {
        const errors = {};
        const fields = ['width', 'height'];
        fields.forEach((field) => {
            if (!couldBeNumber(values[field])) {
                errors[field] = 'Must be a number';
            } else if (Number(values[field]) < 80) {
                errors[field] = 'Value is too small';
            } else if (Number(values[field]) > 5000) {
                errors[field] = 'Value is too big';
            }
        });
        return errors;
    }

    constructor(props) {
        super(props);

        this.popupRef = React.createRef();
        this._stateValueTmp = null;
        this._prevActiveValue = null;
    }

    onCancel = (reset) => {
        reset();
        this.popupRef.current.hide();
    };

    onSubmit = (...rest) => {
        const { onSubmit } = this.props;
        onSubmit && onSubmit(...rest);
        this.popupRef.current.hide();
    };

    onFormStateChange(form, { active, values }) {
        if (!!active && couldBeNumber(values[active]) && this._prevActiveValue !== values[active]) {
            const secondSizeKey = active === 'width' ? 'height' : 'width';
            const calcSecondSize = () => {
                if (values[active] === '') {
                    return '';
                }
                const numValue = Number(values[active]);
                const ratio = this.props[`${secondSizeKey}Init`] / this.props[`${active}Init`];
                return Math.round(numValue * ratio);
            };

            this._stateValueTmp = calcSecondSize();
            this._prevActiveValue = values[active];
            form.mutators[`set_${secondSizeKey}`]();
        }
    }

    /**
     * @public
     */
    show() {
        this.popupRef.current.show();
    }

    renderFormSpy(form) {
        return (
            <FormSpy
                subscription={{
                    active: true,
                    values: true,
                }}
                onChange={this.onFormStateChange.bind(this, form)}
            />
        );
    }

    renderField(fieldKey) {
        return (
            <Field
                name={fieldKey}
                render={({ input, meta }) => (
                    <FormGroup
                        errorText={meta.error}
                    >
                        <label
                            htmlFor={`img-${fieldKey}`}
                        >
                            {fieldKey} (px)
                        </label>
                        <FormInput
                            placeholder={`Enter ${fieldKey}`}
                            id={`img-${fieldKey}`}
                            {...input}
                        />
                    </FormGroup>
                )}
            />
        );
    }

    render() {
        const { widthInit, heightInit } = this.props;
        return (
            <Popup
                title='Resize image'
                ref={this.popupRef}
                showCloseBtn={false}
            >
                <Form
                    initialValues={{
                        width: widthInit,
                        height: heightInit,
                    }}
                    onSubmit={this.onSubmit}
                    mutators={{
                        set_height: (args, state, utils) => {
                            utils.changeValue(state, 'height', () => this._stateValueTmp);
                        },
                        set_width: (args, state, utils) => {
                            utils.changeValue(state, 'width', () => this._stateValueTmp);
                        },
                    }}
                    validate={MIResizePopup.validate}
                    render={({ handleSubmit, invalid, form }) => (
                        <form onSubmit={handleSubmit}>
                            <div className='row'>
                                <div className='col-sm'>
                                    {this.renderField('width')}
                                </div>
                                <div className='col-sm'>
                                    {this.renderField('height')}
                                </div>
                            </div>
                            <PopupButtonsContainer>
                                <FormButtonsRow>
                                    <Button
                                        onClick={this.onCancel.bind(null, form.reset)}
                                        secondary
                                    >
                                        Cancel
                                    </Button>
                                    <Button
                                        type='submit'
                                        disabled={invalid}
                                    >
                                        Resize
                                    </Button>
                                </FormButtonsRow>
                            </PopupButtonsContainer>
                            {this.renderFormSpy(form)}
                        </form>
                    )}
                />
            </Popup>
        );
    }
}

MIResizePopup.propTypes = {
    widthInit: PropTypes.number,
    heightInit: PropTypes.number,
    onSubmit: PropTypes.func,
};

MIResizePopup.defaultProps = {
    widthInit: 0,
    heightInit: 0,
    onSubmit: null,
};

export default MIResizePopup;
