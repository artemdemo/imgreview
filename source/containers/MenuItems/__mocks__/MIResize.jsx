import React from 'react';

const MIResize = props => (
    <div data-mock='MIResize'>
        {props.children}
    </div>
);

export default MIResize;
