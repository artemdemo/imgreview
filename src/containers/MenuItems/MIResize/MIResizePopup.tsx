/* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for */
import React, { useReducer, useEffect } from 'react';
import { FormGroup } from '../../../components/FormGroup/FormGroup';
import FormInput from '../../../components/FormInput/FormInput';
import PopupButtonsContainer from '../../../components/Popup/PopupButtonsContainer';
import FormButtonsRow from '../../../components/FormButtonsRow/FormButtonsRow';
import Button, { EButtonAppearance } from '../../../components/Button/Button';
import Popup from '../../../components/Popup/Popup';
import { couldBeNumber } from '../../../services/number';

type Props = {
  onSubmit: (...rest: any) => void;
  onCancel: () => void;
  widthInit: number;
  heightInit: number;
  show: boolean;
};

type ValuesState = {
  width: number;
  height: number;
  ratioInit: number;
};

type DimensionValues = 'width' | 'height';

function valuesReducer(state: ValuesState, action: any) {
  switch (action.type) {
    case 'width':
      return {
        ...state,
        width: action.data,
        // height: Math.round(Number(action.data) * state.ratioInit),
      };
    case 'height':
      return {
        ...state,
        height: action.data,
        // width: Math.round(Number(action.data) * state.ratioInit),
      };
    case 'ratioInit':
      return {
        ...state,
        ratioInit: action.data,
      };
    default:
      return state;
  }
}

type ErrorsState = {
  widthError: string;
  heightError: string;
};

type ErrorValues = 'widthError' | 'heightError';

function errorsReducer(state: ErrorsState, action: any) {
  switch (action.type) {
    case 'widthError':
      return {
        ...state,
        widthError: action.data,
      };
    case 'heightError':
      return {
        ...state,
        heightError: action.data,
      };
    default:
      return state;
  }
}

const MIResizePopup: React.FC<Props> = (props) => {
  const { onSubmit, onCancel, widthInit, heightInit, show } = props;
  const [valuesState, dispatchValues] = useReducer(valuesReducer, {
    width: 0,
    height: 0,
    ratioInit: 1,
  });
  const [errorsState, dispatchErrors] = useReducer(errorsReducer, {
    widthError: '',
    heightError: '',
  });

  useEffect(() => {
    if (show) {
      dispatchValues({ type: 'width', data: widthInit });
      dispatchValues({ type: 'height', data: heightInit });
      dispatchValues({ type: 'ratioInit', data: widthInit / heightInit });
    }
  }, [show]);

  useEffect(() => {
    const errors: { [key: string]: string } = {};
    const fields = ['width', 'height'];
    fields.forEach((field: string) => {
      if (!couldBeNumber(valuesState[field as DimensionValues])) {
        errors[field] = 'Must be a number';
      } else if (Number(valuesState[field as DimensionValues]) < 80) {
        errors[field] = 'Value is too small';
      } else if (Number(valuesState[field as DimensionValues]) > 5000) {
        errors[field] = 'Value is too big';
      }
    });
    dispatchErrors({ type: 'widthError', data: errors.width || '' });
    dispatchErrors({ type: 'heightError', data: errors.height || '' });
  }, [valuesState]);

  const renderField = (fieldKey: string) => {
    return (
      <FormGroup errorText={errorsState[`${fieldKey}Error` as ErrorValues]}>
        <label htmlFor={`img-${fieldKey}`}>{fieldKey} (px)</label>
        <FormInput
          placeholder={`Enter ${fieldKey}`}
          value={valuesState[fieldKey as DimensionValues]}
          onChange={(e) => {
            const { value } = e.target;
            dispatchValues({ type: fieldKey, data: value });
          }}
          id={`img-${fieldKey}`}
        />
      </FormGroup>
    );
  };

  return (
    <Popup
      title="Resize image"
      onClose={onCancel}
      show={show}
      showCloseBtn={false}
    >
      <form
        onSubmit={(e) => {
          e.preventDefault();
          onSubmit({ width: valuesState.width, height: valuesState.height });
        }}
      >
        <div className="row">
          <div className="col-sm">{renderField('width')}</div>
          <div className="col-sm">{renderField('height')}</div>
        </div>
        <PopupButtonsContainer>
          <FormButtonsRow>
            <Button onClick={onCancel} appearance={EButtonAppearance.SECONDARY}>
              Cancel
            </Button>
            <Button
              type="submit"
              disabled={
                errorsState.widthError !== '' || errorsState.heightError !== ''
              }
            >
              Resize
            </Button>
          </FormButtonsRow>
        </PopupButtonsContainer>
      </form>
    </Popup>
  );
};

export default MIResizePopup;
