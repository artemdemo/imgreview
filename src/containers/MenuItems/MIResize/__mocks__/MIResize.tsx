import React from 'react';

const MIResize = (props: any) => (
  <div data-mock="MIResize">{props.children}</div>
);

export default MIResize;
