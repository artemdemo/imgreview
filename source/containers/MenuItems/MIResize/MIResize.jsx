/* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for,react/no-unused-state */
import React from 'react';
import { connect } from 'react-redux';
import Icon from '../../../components/Icon/Icon';
import MainMenuItem from '../../../components/MainMenu/MainMenuItem';
import MIResizePopup from './MIResizePopup';
import { updateImageSize } from '../../../model/canvas/canvasActions';

class MIResize extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            width: 0,
            height: 0,
        };

        this.popupRef = React.createRef();

        this.widthInit = 0;
        this.heightInit = 0;
    }

    onClick = () => {
        const { canvas } = this.props;
        const { width, height } = canvas.image.getSize();
        this.widthInit = width;
        this.heightInit = height;
        this.setState({
            width,
            height,
        }, () => {
            this.popupRef.current.show();
        });
    };

    onSubmit = (values) => {
        const { updateImageSize } = this.props;
        const width = Number(values.width);
        const height = Number(values.height);
        if (width > 0 && height > 0) {
            updateImageSize({width, height});
        }
    };

    render() {
        const { canvas } = this.props;
        return (
            <React.Fragment>
                <MainMenuItem
                    onClick={this.onClick}
                    disabled={canvas.image == null}
                >
                    <Icon
                        name='expand'
                        title='Resize'
                    />
                </MainMenuItem>
                <MIResizePopup
                    widthInit={this.state.width}
                    heightInit={this.state.height}
                    onSubmit={this.onSubmit}
                    ref={this.popupRef}
                />
            </React.Fragment>
        );
    }
}

export default connect(
    state => ({
        canvas: state.canvas,
    }), {
        updateImageSize,
    },
)(MIResize);
