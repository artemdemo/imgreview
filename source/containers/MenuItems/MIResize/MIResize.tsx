/* eslint-disable jsx-a11y/label-has-associated-control,jsx-a11y/label-has-for,react/no-unused-state */
import React from 'react';
import Icon from '../../../components/Icon/Icon';
import TopMenuItem from '../../../components/TopMenu/TopMenuItem';
import MIResizePopup from './MIResizePopup';
import { updateCanvasSize } from '../../../canvas/api';

type Props = {
    disabled: boolean;
};

type State = {
    width: number;
    height: number;
};

class MIResize extends React.PureComponent<Props, State> {
    private readonly popupRef: any;

    static readonly defaultProps = {
        disabled: false,
    };

    constructor(props) {
        super(props);

        this.state = {
            width: 0,
            height: 0,
        };

        this.popupRef = React.createRef();
    }

    onClick = () => {
        const { canvas } = this.props;
        const { width, height } = canvas.image.getSize();
        this.setState({
            width,
            height,
        }, () => {
            this.popupRef.current.show();
        });
    };

    onSubmit = (values) => {
        const width = Number(values.width);
        const height = Number(values.height);
        if (width > 0 && height > 0) {
            updateCanvasSize({width, height});
        }
    };

    render() {
        const { disabled } = this.props;
        return (
            <React.Fragment>
                <TopMenuItem
                    onClick={this.onClick}
                    disabled={disabled}
                >
                    <Icon
                        name='expand'
                        title='Resize'
                    />
                </TopMenuItem>
                <MIResizePopup
                    widthInit={this.state.width}
                    heightInit={this.state.height}
                    onSubmit={this.onSubmit}
                    ref={this.popupRef}
                />
            </React.Fragment>
        );
    }
}

export default MIResize;
