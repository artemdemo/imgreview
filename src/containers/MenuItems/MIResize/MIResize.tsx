import React, { useState } from 'react';
import { TopMenuItem } from '../../../components/TopMenu/TopMenuItem';
import MIResizePopup from './MIResizePopup';
import { t } from '../../../services/i18n';
import { EIcon, ImgIcon } from '../ImgIcon/ImgIcon';

type Props = {
  disabled?: boolean;
};

export const MIResize: React.FC<Props> = (props) => {
  const { disabled } = props;
  const [showPopup, setShowPopup] = useState(false);

  const onSubmit = (values: { width: string; height: string }) => {
    console.error(
      `This API is no longer supported, since canvas is nov infinite.\nI can't resize infinite canvas.`,
    );
    // const width = Number(values.width);
    // const height = Number(values.height);
    // if (width > 0 && height > 0) {
    //   setWidth(0);
    //   setHeight(0);
    //   setShowPopup(false);
    //   canvasApi.updateCanvasSize({ width, height });
    // }
  };

  return (
    <>
      <TopMenuItem
        onClick={() => {
          setShowPopup(true);
        }}
        disabled={disabled}
        title={t('menu.resize')}
        stopPropagation={false}
      >
        <ImgIcon icon={EIcon.resize} />
      </TopMenuItem>
      <MIResizePopup
        widthInit={0}
        heightInit={0}
        onSubmit={onSubmit}
        onCancel={() => {
          setShowPopup(false);
        }}
        show={showPopup}
      />
    </>
  );
};
