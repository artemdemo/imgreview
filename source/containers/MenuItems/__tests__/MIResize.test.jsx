import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MIResize from '../MIResize';

jest.mock('react-redux');
jest.mock('react-hotkeys');
jest.mock('../../../canvas/Arrow/Arrow');
jest.mock('../../../components/Popup/Popup');
jest.mock('../../../components/Icon/Icon');
jest.mock('../../../components/MainMenu/MainMenuItem');

const emptyState = {
    canvas: {
        image: null,
    },
};

describe('MIResize', () => {
    jest.clearAllMocks();
    const reactReduxMock = require('react-redux');

    it('should render', () => {
        const tree = renderer.create(
            <MIResize
                canvas={emptyState.canvas}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
        expect(reactReduxMock.__getLastMaps().mapStateToProps(emptyState)).toEqual({
            canvas: emptyState.canvas,
        });
    });

    it('should trigger popup open', () => {
        const wrapper = mount(
            <MIResize
                canvas={emptyState.canvas}
            />
        );
        const instance = wrapper.instance();
        const showMock = jest.fn();
        instance.popupRef = {
            current: {
                show: showMock,
            },
        };
        instance.onClick();
        expect(showMock).toBeCalled();
    });

    it('should handle popup open', () => {
        const width = 10;
        const height = 20;
        const state = {
            canvas: {
                image: {
                    getSize: () => ({
                        width,
                        height,
                    }),
                },
            },
        };
        const wrapper = mount(
            <MIResize
                canvas={state.canvas}
            />
        );
        const instance = wrapper.instance();
        const setStateMock = jest.fn();
        instance.setState = setStateMock;
        instance.onPopupOpen();
        expect(setStateMock).toBeCalledWith(expect.objectContaining({
            width,
            height,
            widthInit: width,
            heightInit: height,
        }));
    });

    it('should handle onResize', () => {
        const width = '10';
        const height = '20';
        const updateImageSizeMock = jest.fn();
        const wrapper = mount(
            <MIResize
                canvas={emptyState.canvas}
                updateImageSize={updateImageSizeMock}
            />
        );
        wrapper.setState({
            width,
            height,
        });
        const instance = wrapper.instance();
        instance.onResize();
        expect(updateImageSizeMock).toBeCalledWith({
            width: Number(width),
            height: Number(height),
        });
    });

    describe('should handle updateSize', () => {
        const value = '10';
        const e = {
            target: {
                value,
            },
        };
        const wrapper = mount(
            <MIResize
                canvas={emptyState.canvas}
            />
        );
        wrapper.setState({
            widthInit: 10,
            heightInit: 5,
        });
        const instance = wrapper.instance();
        const setStateSpy = jest.spyOn(instance, 'setState');

        it('width', () => {
            jest.clearAllMocks();
            instance.updateSize('width', e);
            expect(setStateSpy).toBeCalledWith({
                height: 5,
                width: '10',
            });
        });

        it('height', () => {
            jest.clearAllMocks();
            instance.updateSize('height', e);
            expect(setStateSpy).toBeCalledWith({
                width: 20,
                height: '10',
            });
        });

        it('should not call setState', () => {
            jest.clearAllMocks();
            const _e = {
                target: {
                    value: 'smth',
                },
            };
            instance.updateSize('height', _e);
            expect(setStateSpy).not.toBeCalled();
        });
    });
});
