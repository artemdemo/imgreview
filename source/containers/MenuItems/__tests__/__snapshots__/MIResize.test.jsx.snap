// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIResize should render 1`] = `
Array [
  <div
    data-mock="MainMenuItem"
    onClick={[Function]}
  >
    <div
      data-mock="Icon"
    />
  </div>,
  <div
    data-mock="Popup"
  >
    <div
      className="row"
    >
      <div
        className="col-sm"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="img-width"
          >
            Width (px)
          </label>
          <div
            data-mock="HotKeys"
          >
            <input
              className="form-input"
              id="img-width"
              onChange={[Function]}
              onSubmit={[Function]}
              placeholder="Enter width"
              value="0"
            />
          </div>
        </div>
      </div>
      <div
        className="col-sm"
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="img-height"
          >
            Height (px)
          </label>
          <div
            data-mock="HotKeys"
          >
            <input
              className="form-input"
              id="img-height"
              onChange={[Function]}
              onSubmit={[Function]}
              placeholder="Enter height"
              value="0"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
